src/containers/maps/MapVisuals.jsx

```diff
@@ -0,0 +1,171 @@
+      if (index === 0) tl.addTo(this.map);  //Use the first tile layer as the default tile layer.
+    });
+    //--------------------------------
+    
+    //Prepare the dynamic data layer.
+    //Starts off empty, but is populated by updateDataLayer().
+    //--------------------------------
+    this.dataLayer = L.geoJson(null, {
+      pointToLayer: this.renderMarker
+    }).addTo(this.map);
+    Actions.getMapMarkers(this.props.mapConfig);
+    //--------------------------------
+    
+    //Prepare additional geographic information layers (park boundaries, etc)
+    //--------------------------------
+    const geomapLayers = {};
+    //--------------------------------
+    
+    //Add standard 'Layer' controls
+    //--------------------------------
+    L.control.layers(tileLayers, {
+      'Data': this.dataLayer,
+      //...geomapLayers
+    }, {
+      position: 'topleft',
+      collapsed: true,
+    }).addTo(this.map);
+    //--------------------------------
+  }
+  
+  //----------------------------------------------------------------
+  
+  updateDataLayer(props = this.props) {
+    if (!this.map || !this.dataLayer || !props.markersData) return;
+    
+    this.dataLayer.clearLayers();
+    this.dataLayer.addData(props.markersData);  //Markers Data must be in GeoJSON format.
+  }
+  
+  renderMarker(feature, latlng) {
+    const count = (feature && feature.properties && feature.properties.count !== undefined)
+      ? feature.properties.count : 0;  //Warning: assumption is `count` is an integer.
+    const radius = 
+      Math.max(Math.min((count - DEFAULT_MARKER.minValue) / DEFAULT_MARKER.maxValue, 1), 0) *
+      (DEFAULT_MARKER.maxRadius - DEFAULT_MARKER.minRadius) +
+      DEFAULT_MARKER.minRadius;
+    
+    const marker = L.circleMarker(latlng, {
+      color: DEFAULT_MARKER.color,
+      fillColor: (count > 0) ? DEFAULT_MARKER.fillColor : DEFAULT_MARKER.emptyFillColor,
+      fillOpacity: DEFAULT_MARKER.fillOpacity,
+      radius,
+    });
+    
+    marker.on('click', this.examineMarker);
+    return marker;
+  }
+  
+  examineMarker(e) {
+    if (!e || !e.target || !e.target.feature || !e.target.feature.properties) return;
+    
+    const cameraId = e.target.feature.properties.id;
+    alert(`TODO: Examine camera ${cameraId}`);  //TODO: Make this a "View a single camera's photos" action.
+  }
+  
+  //----------------------------------------------------------------
+  
+  render() {
+    return (
+      <section ref="mapVisuals" className="map-visuals">
+        <link
+          rel="stylesheet" href="https://unpkg.com/leaflet@1.2.0/dist/leaflet.css"
+          integrity="sha512-M2wvCLH6DSRazYeZRIm1JnYyh22purTM+FDB5CsyxtQJYeKq83arPe5wgbNmcFXGqiSH2XR8dT/fJISVA1r/zQ=="
+          crossOrigin=""
+        />
+        <div id="mapVisuals"></div>
+      </section>
+    );
+  }
+  
+  componentDidMount() {
+    this.initMapExplorer();
+  }
+  
+  componentWillReceiveProps(nextProps) {
+    this.updateDataLayer(nextProps);
```

Comment:
That's correct, it needs to be fired every time `this.props.markersData` is updated. This is because `updateDataLayer()` is the equivalent of a `render` method for the implementation of the Leaflet map.

The map doesn't natively listen to React prop changes natively, so this tells the map when to react. (This might be something `react-leaflet` solves, but I'd need to explore that implementation before I can comment confidently.)